salesforce_functions/data_api/requests.py

```diff
@@ -0,0 +1,287 @@
+from typing import Any, Generic, Literal, TypeVar, cast
+from urllib.parse import urlencode
+
+from multidict import CIMultiDictProxy
+
+from .exceptions import (
+    InnerSalesforceRestApiError,
+    MissingIdFieldError,
+    SalesforceRestApiError,
+    UnexpectedRestApiResponsePayload,
+)
+from .record import QueriedRecord, Record, RecordQueryResult
+from .reference_id import ReferenceId
+
+HttpMethod = Literal["GET", "POST", "PATCH", "DELETE"]
+Json = dict[str, Any] | list[Any]
+
+T = TypeVar("T")
+
+
+class RestApiRequest(Generic[T]):
```

Comment:
The `salesforce_functions.data_api.requests` module is currently public, whereas I presume we don't want to expose it, since it's mostly an implementation detail?

We could rename the module to `_requests` though that then means we have private things under `_internal` and separately under `data_api`. I would need to think more about what's best here.

Separately, I think we should consider re-exporting some of the public types (eg `Record` and so on) from a higher level module, to simplify the importing of them.

I think all of the above can wait until after this PR - so I've filed GUS-W-12099003 to track.

Reply:
So thinking about this more overnight it seems to make sense to keep all the DataAPI things together - and so not put anything under `_internal`. And then rename `data_api/requests.py` to `data_api/_requests.py` here.

Reply:
Renamed in b7e557ea45883fa9d44b199be19fe745c4ffbe7e

Reply:
That makes sense to me. It's totally not part of the public API. 👍🏻